src/rgw/driver/sfs/sqlite/dbconn.cc

```diff
@@ -414,4 +416,24 @@ void DBConn::maybe_upgrade_metadata() {
   }
 }
 
+void DBConn::maybe_rename_database_file() const {
```

Comment:
I suspect we may also have to rename the `-wal` and `-shm` files. I think those can be left behind if the database is not properly closed.

Reply:
Yeah, if you `killall -9 radosgw` the `-wal` and `-shm` files will remain, then the next time you start it up, you'll see something like `[SQLITE] (283) recovered 26 frames from WAL file /scratch/s3gw/qa/s3gw.db-wal` in the log. So _if_ we keep the migration code, we'd need to migrate those two as well.

Reply:
Thinking about it, the rename code is a bit naive. Perhaps too naive. Not only do we need to rename the extra files, there may also be temporary files (according to docs) that share the basename. If the database is still open renaming is also a big mistake.

A safer option would be the backup API - I think we should rather use that